shell/platform/darwin/ios/ios_surface_metal_impeller.mm

```diff
@@ -4,10 +4,75 @@
 
 #import "flutter/shell/platform/darwin/ios/ios_surface_metal_impeller.h"
 
+#include "flutter/impeller/renderer/backend/metal/formats_mtl.h"
+#include "flutter/impeller/renderer/context.h"
 #include "flutter/shell/gpu/gpu_surface_metal_impeller.h"
 
+namespace impeller {
```

Comment:
Nit: Could we remove the use of the Impeller namespace here, since symbols being defined here (`CustomColorAttachmentPixelFormatContext`) are not part of Impeller.

Reply:
If I do that the code is going to explode.  There are tons of symbols from `::impeller` in it.  FWIW this class isn't really in `::impeller`.  It's in `::impeller::(anonymous namespace)`.  Once you are in an an anonymous namespace the other parts of your namespace are inconsequential except for having convenient access to symbols.  I could pull this out of the anonymous namespace and into its own file, but considering how we want to fix the issue that lead to us needing this, that's probably overklll. 

I've filed an issue to try to capture this: https://github.com/flutter/flutter/issues/120499